diff --git a/medialoader/sound-libs/mpg123/src/compat/compat.c b/medialoader/sound-libs/mpg123/src/compat/compat.c
--- a/medialoader/sound-libs/mpg123/src/compat/compat.c
+++ b/medialoader/sound-libs/mpg123/src/compat/compat.c
@@ -500 +500 @@
-	new_sa.sa_handler = handler;
+	new_sa.sa_handler = (void (*)(int))handler;
@@ -505 +505 @@
-	return (old_sa.sa_handler);
+	return ((void (*)())old_sa.sa_handler);
