diff --git a/squirrel/squirrel/CMakeLists.txt b/squirrel/squirrel/CMakeLists.txt
index f96b958..cf2d1c5 100644
--- a/squirrel/squirrel/CMakeLists.txt
+++ b/squirrel/squirrel/CMakeLists.txt
@@ -41,5 +41,9 @@ if(NOT DISABLE_STATIC)
   target_include_directories(squirrel_static PUBLIC
     "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
     "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
     )
+  if(MSVC)
+    target_compile_definitions(squirrel_static PRIVATE
+      "SQUIRREL_API=__declspec(dllexport)")
+  endif()
 endif()
diff --git a/squirrel/sqstdlib/CMakeLists.txt b/squirrel/sqstdlib/CMakeLists.txt
index 34fba82..3d7c3b9 100644
--- a/squirrel/sqstdlib/CMakeLists.txt
+++ b/squirrel/sqstdlib/CMakeLists.txt
@@ -38,5 +38,9 @@ if(NOT DISABLE_STATIC)
   target_include_directories(sqstdlib_static PUBLIC
     "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
     "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
     )
+  if(MSVC)
+    target_compile_definitions(sqstdlib_static PRIVATE
+      "SQUIRREL_API=__declspec(dllexport)")
+  endif()
 endif()
diff --git a/simplesquirrel/CMakeLists.txt b/simplesquirrel/CMakeLists.txt
index 227938d..50b3c2f 100644
--- a/simplesquirrel/CMakeLists.txt
+++ b/simplesquirrel/CMakeLists.txt
@@ -58,2 +58,5 @@
 # Install
+if(MSVC)
+  target_compile_definitions(${PROJECT_NAME}_static PRIVATE SSQ_EXPORTS=1 SSQ_DLL=1)
+endif()
 if(BUILD_INSTALL)
