diff --git a/include/zeroerr/unittest.h b/include/zeroerr/unittest.h
index 32ac9ea..b9efbe5 100644
--- a/include/zeroerr/unittest.h
+++ b/include/zeroerr/unittest.h
@@ -44,8 +44,9 @@ ZEROERR_SUPPRESS_COMMON_WARNINGS_PUSH
     inline void classname::funcname(ZEROERR_UNUSED(zeroerr::TestContext* _ZEROERR_TEST_CONTEXT))
 
 #define TEST_CASE_FIXTURE(fixture, ...)                                              \
-    ZEROERR_EXPAND(ZEROERR_CREATE_TEST_CLASS(fixture, ZEROERR_NAMEGEN(_zeroerr_class), \
-                              ZEROERR_NAMEGEN(_zeroerr_test_method), __VA_ARGS__))
+    ZEROERR_EXPAND(ZEROERR_CREATE_TEST_CLASS(                                         \
+        fixture, ZEROERR_NAMEGEN(ZEROERR_CAT(fixture, _zeroerr_class)),               \
+        ZEROERR_NAMEGEN(ZEROERR_CAT(fixture, _zeroerr_test_method)), __VA_ARGS__))
 
 
 #define ZEROERR_HAVE_SAME_OUTPUT _ZEROERR_TEST_CONTEXT->save_output();
